pkg/gen/input/workflows/internal/file/create_release.go

```diff
@@ -204,6 +204,62 @@ jobs:
           tag_name: "v${{ needs.gather_facts.outputs.version }}"
           release_name: "v${{ needs.gather_facts.outputs.version }}"
 
+  create-release-branch:
+    name: Create release branch
+    runs-on: ubuntu-18.04
+    needs:
+      - gather_facts
+      - install_semver
+    if: ${{ needs.gather_facts.outputs.version }}
+    steps:
+      - name: Download semver artifact to /opt/bin
+        uses: actions/download-artifact@v2
+        with:
+          name: semver
+          path: /opt/bin
+      - name: Prepare /opt/bin
+        run: |
+          chmod +x /opt/bin/*
+          echo "::add-path::/opt/bin"
+      - uses: actions/checkout@v2
+        name: Check out the repository
+        with:
+          fetch-depth: 0  # Clone the whole history, not just the most recent commit.
+      - name: Fetch all tags and branches
+        run: "git fetch --all"
+      - name: Create long-lived release branch
+        run: |
+          current_version="${{ needs.gather_facts.outputs.version }}"
+          parent_version="$(git describe --tags --abbrev=0 HEAD^)"
+          parent_version="${parent_version#v}" # Strip "v" prefix.
+          echo "current_version=$current_version parent_version=$parent_version"
+
+          current_major=$(semver get major $current_version)
+          current_minor=$(semver get minor $current_version)
+          parent_major=$(semver get major $parent_version)
+          parent_minor=$(semver get minor $parent_version)
+          echo "current_major=$current_major current_minor=$current_minor parent_major=$parent_major parent_minor=$parent_minor"
+
+          if [[ $current_major -gt $parent_major ]] ; then
+            echo "Current tag is a new major version"
+          elif [[ $current_major -eq $parent_major ]] && [[ $current_minor -gt $parent_minor ]] ; then
+            echo "Current tag is a new minor version"
+          else
+            echo "Current tag is not a new major or minor version. Nothing to do here."
+            exit 0
+          fi
+
+          release_branch="release-v${parent_major}.${parent_minor}.x"
+          echo "release_branch=$release_branch"
+
+          if git rev-parse --verify master $release_branch ; then
```

Comment:
Last minute fix. I feel like we can remove this `if` altogether though. In the worst case scenario `git push` will fail and if it fails it's a good thing because we'll be notified that release branch head points at wrong commit.

```suggestion
          if git rev-parse --verify $release_branch ; then
```

Reply:
I think it's nice to have the explicit feedback, but we could make it a fail if the branch already exists. I don't think it's out of the question that we'd need to tag some branch besides master for a hotfix or x.x.x-0 release which could trigger this workflow. My only hesitation to make it fail is that it would then break downstream CI in such a case, which would be annoying in an already annoying situation. WDYT?

Reply:
This job has no dependency, so the the action would be red, but release would be created anyway. I thought it would somewhat increase visibility if we have a conflict. If the branch already exists and points to the correct head then we'd only have "Already up to date" or something similar.

I leave the decision to you. We can iterate.

Reply:
Ok, I'll go with this and we can revisit if needed